libraries/AP_HAL_ChibiOS/hwdef/common/stm32_util.c

```diff
@@ -468,3 +468,39 @@ uint32_t stack_free(void *stack_base)
     return ((uint32_t)p) - (uint32_t)stack_base;
 }
 #endif
+
+#ifdef RNG
+static uint32_t stm32_rand_generate(void)
+{
+    static uint32_t last_value = 0;
+    static uint32_t new_value = 0;
+    uint32_t error_bits = 0;
+    error_bits = RNG_SR_SEIS | RNG_SR_CEIS;
+    while (new_value == last_value) {
```

Comment:
should be do {} while() not while ()

Reply:
ChibiOS changes moved to separate PR https://github.com/ArduPilot/ardupilot/pull/17855